pkg/felt/other.go

```diff
@@ -0,0 +1,150 @@
+package felt
+
+import (
+	"github.com/NethermindEth/juno/pkg/common"
+)
+
+var (
+	Felt0 = new(Felt).SetZero()
+	Felt1 = new(Felt).SetOne()
+	Felt2 = new(Felt).SetUint64(2)
+	Felt3 = new(Felt).SetUint64(3)
+)
+
+// ByteSlice() is exactly the same as Marshal() except that it returns
+// null when z is null. This mimics the behavior of big.Int.
+func (z *Felt) ByteSlice() []byte {
```

Comment:
Let's rename this to [`Bytes()`](https://pkg.go.dev/math/big#Int.Bytes) for consistency not only with `big.Int` but with the rest of the standard library i.e. [`Buffer.Bytes()`](https://pkg.go.dev/bytes@go1.18.4#Buffer.Bytes).

Reply:
I agree this is preferable, but `Bytes()` is [already defined](https://github.com/NethermindEth/juno/blob/fe2c5e5b5eac5dafab577922a62e5a34b8f2bd8b/pkg/felt/felt.go#L848-L856). It has some inconvenient properties: it returns an array and crashes when the caller is nil (unlike `big.Int`). `felt.Marshal` returns a byte slice, but crashes when caller is nil.

Reply:
Ah. I see.